cg-spec/editors_draft.html

```diff
@@ -1420,6 +1420,112 @@ <h2>isTRIPLE</h2>
+             class="nohighlight example"
+        >
+          <!--
+          PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
+          PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
+
+          SELECT ?t WHERE {
+            ?s ?p ?o .
+            BIND( TRIPLE(?s,rdf:type,rdfs:Resource) AS ?t )
+          }
+          -->
+        </pre>
+
+        <p>There are cases in which the effects of using the new TRIPLE function are not immediately obvious just from its definition itself. In particular, this concerns the interplay of this function with the <a data-cite="SPARQL11-QUERY#func-bnode">BNODE function</a> when used within <a data-cite="SPARQL11-QUERY#queryDataset">GRAPH clauses</a>. In the following, we discuss such cases.</p>
+
+        <p>Consider the following SPARQL-star query.</p>
+
+        <pre data-transform="updateExample"
+             data-content-type="application/x-sparql-star-query"
+             class="nohighlight example"
+        >
+          <!--
+          SELECT ( COUNT(?t1) AS ?t1Count )
+                 ( COUNT(?t2) AS ?t2Count )
+                 ( COUNT(?t3) AS ?t3Count )
+                 ( COUNT(DISTINCT ?t1) AS ?t1DistCount )
+                 ( COUNT(DISTINCT ?t2) AS ?t2DistCount )
+                 ( COUNT(DISTINCT ?t3) AS ?t3DistCount )
+          WHERE {
+            GRAPH ?g {
+              BIND( TRIPLE(BNODE(), :p, :o) AS ?t1 )
+              BIND( TRIPLE(BNODE("id"), :p, :o) AS ?t2 )
+              BIND( TRIPLE(:s, :p, :o) AS ?t3 )
+            }
+          }
+          -->
+        </pre>
+
+        <p>When evaluated over an <a>RDF-star dataset</a> that contains |n| <a>named graphs</a>, with |n|>0, the GRAPH clause results in one <a>SPARQL-star solution mapping</a> per named graph. The SELECT clause collapses these |n| mappings into a single mapping that is defined for the six variables introduced in the SELECT clause (if |n|=0, the query result is the empty set). Each of these six variables is mapped to an integer-typed literal with the following values. For <code>?t1Count</code>, <code>?t2Count</code>, and <code>?t3Count</code>, the value is&nbsp;|n|, respectively, because the variables <code>?t1</code>, <code>?t2</code>, and <code>?t3</code> are bound in each of the |n| solution mappings produced for the GRAPH clause. For the variable&nbsp;<code>?t1DistCount</code>, the value is |n| as well. The reason for the latter is that the BIND clauses are evaluated |n| times (once for each named graph) and the subexpression BNODE() always creates a different new blank node; thus, the triples created for variable <code>?t1</code> are all different from one another. The same is also the case if the BNODE function is used with an argument; i.e., the value of <code>?t2DistCount</code> is |n| too. In contrast, the variable&nbsp;<code>?t3</code> is mapped to the same triple within each evaluation of the third BIND clause and, thus, the value for <code>?t3DistCount</code> is&nbsp;1.</p>
```

Comment:
```suggestion
        <p>When evaluated over an <a>RDF-star dataset</a> that contains |n| <a>named graphs</a>, where |n|>0, the GRAPH clause results in one <a>SPARQL-star solution mapping</a> per named graph. The SELECT clause collapses these |n| mappings into a single mapping that is defined for the six variables introduced in the SELECT clause. (If |n|=0, the query result is the empty set.) Each of these six variables is mapped to an integer-typed literal with values as follows. For each of <code>?t1Count</code>, <code>?t2Count</code>, and <code>?t3Count</code>, the value is&nbsp;|n|, because the variables <code>?t1</code>, <code>?t2</code>, and <code>?t3</code> are bound in each of the |n| solution mappings produced for the GRAPH clause. For the variable&nbsp;<code>?t1DistCount</code>, the value is also |n|, because the BIND clauses are evaluated |n| times (once for each named graph) and the subexpression BNODE() always creates a different new blank node; thus, the triples created for variable <code>?t1</code> are all different from one another. This is also the case if the BNODE function is used with an argument; i.e., the value of <code>?t2DistCount</code> is |n| too. In contrast, the variable&nbsp;<code>?t3</code> is mapped to the same triple within each evaluation of the third BIND clause; thus, the value for <code>?t3DistCount</code> is&nbsp;1.</p>
```